File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        Queue<Integer> queue = new PriorityQueue<>();
        int minScoville = Integer.MAX_VALUE;
        for (int i = 0; i < scoville.length; i++) {
            queue.offer(scoville[i]);
            if (minScoville > scoville[i]) {
                minScoville = scoville[i];
            }
        }
        
        if (minScoville >= K) {
            return 0;
        }
        
        while(!(minScoville >= K) && queue.size() > 2) {
            int food1 = queue.poll();
            int food2 = queue.poll();
            int newScoville = food1 + 2 * food2;
            int nextFoodScoville = queue.peek();
            minScoville = Math.min(newScoville, nextFoodScoville);
            answer++;
            queue.offer(newScoville);
        }
        
        if (minScoville >= K) {
            return answer;
        } else {
            int food1 = queue.poll();
            int food2 = queue.poll();
            int newScoville = food1 + 2 * food2;
            if (newScoville >= K) {
                return answer + 1;
            } else {
                return -1;
            }
        }
    }
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    arr = []
    for s in scoville:
        heapq.heappush(arr, s)
    
    while len(arr) > 1:
        if arr[0] >= K:
            break
        
        answer += 1
        min1 = heapq.heappop(arr)
        min2 = heapq.heappop(arr)
        
        new_food = min1 + (min2 * 2)
        heapq.heappush(arr, new_food)
        
    if arr[0] < K:
        return -1
    return answer